src/views/AddRecipe.vue

```diff
@@ -0,0 +1,151 @@
+                        </b-form-group>
+                        <br />
+                        <p class="text-danger"> {{ error }} </p>
+                    </b-col>
+                    <b-col cols="12" lg="6">
+                        <b-form-group id="recipeDifficultyGroup"
+                                        v-bind:label="$t('addRecipe.difficulty')"
+                                        label-for="recipeDifficulty">
+                            <b-form-input type="number"
+                                            value="1"
+                                            min="1"
+                                            max="5"
+                                            v-model="difficulty"
+                                            required>
+                            </b-form-input>
+                        </b-form-group>
+                        <br />
+                        <p class="text-danger"> {{ error }} </p>
+                    </b-col>
+                    <b-col cols="12" lg="12">
+                        <b-form-group v-bind:label="$t('addRecipe.instructions')"
+                                        label-for="instructions">
+                            <b-textarea id="instructions"
+                                        v-model="instructions"
+                                        required>
+                            </b-textarea>
+                        </b-form-group>
+                        <br />
+                        <p class="text-danger"> {{ error }} </p>
+                    </b-col>
+                </b-row>
+                <b-button type="submit" variant="warning">{{ $t('addRecipe.add') }}</b-button>
+            </b-form>
+            <br />
+        </b-container>
+    </div>
+</template>
+
+<style scoped>
+    #pagedesc {
+        margin: 20px;
+    }
+</style>
+
+<script>
+import defaultRecipes from '@/data/defaultRecipes.js';
+import categories from '@/data/categories.js'
+
+    export default {
+        components: {
+        },
+        data: function () {
+            return {
+                name: "",
+                type: "",
+                error: "",
+                instructions: "",
+                duration: "",
+                difficulty: "",
+                loggedUserId: null,
+                loggedUserUsername: null,
+                categories: categories
+            };
+        },
+        created() {
+            this.loggedUserId = localStorage.getItem('loggedUserId');
+            this.loggedUserUsername = localStorage.getItem('loggedUserUsername');
+        },
+        methods: {
+            recipeSubmit(event) {
+                event.preventDefault();
+
+                let allRecipes = defaultRecipes;
+                if (localStorage.getItem("recipes") != null)
+                    allRecipes = JSON.parse(localStorage.getItem("recipes"));
+
+                let lastId = Math.max(...allRecipes.map(e => e.id));
```

Comment:
I like this.

Reply:
znam baki to sam specijalno za tebe napisao